env0/utils.go

```diff
@@ -221,6 +221,94 @@ func writeResourceData(i interface{}, d *schema.ResourceData) error {
 	return nil
 }
 
+func getInterfaceValues(i interface{}) []interface{} {
+	var result []interface{}
+
+	val := reflect.ValueOf(i)
+
+	for i := 0; i < val.Len(); i++ {
+		field := val.Index(i)
+		result = append(result, field.Interface())
+	}
+
+	return result
+}
+
+// Extracts values from a slice of interfaces, and writes it to resourcedata at name.
+func writeResourceDataSlice(i interface{}, name string, d *schema.ResourceData) error {
```

Comment:
Why do we need a different func for slice? Can't we simply utilize the existing `writeResourceData`?

Reply:
The end-goal is for writeResourceData to call this function.
I'll try and refactor it a little.

Reply:
Applied some refactoring to make it more readable. Also incorporated the new function in writeResourceData